src/phoenix/server/api/types/Dataset.py

```diff
@@ -17,11 +18,19 @@
 
 @strawberry.type
 class Dataset:
-    name: str = strawberry.field(description="The given name of the dataset")
     start_time: datetime = strawberry.field(description="The start bookend of the data")
     end_time: datetime = strawberry.field(description="The end bookend of the data")
     dataset: strawberry.Private[ms.Dataset]
 
+    # type ignored here to get around the following: https://github.com/strawberry-graphql/strawberry/issues/1929
+    @strawberry.field(description="Returns a human friendly name for the dataset.")  # type: ignore
+    def name(self) -> str:
+        ds_name = self.dataset.name
+        if ds_name.startswith(GENERATED_NAME_PREFIX):
+            # The generated names are UUIDs so use the role as the name
+            return "primary" if self.dataset.role is DatasetRole.PRIMARY else "reference"
+        return ds_name
```

Comment:
nit. maybe we should encapsulate all this in a `display_name` attribute of the `dataset` class, because it looks like implementation details that strawberry (as as client) shouldn't need to worry about
```suggestion
        return self.dataset.display_name
```

Reply:
Yeah I agree with you. I think I wasn't sure if the dataset knew it's role but let me try.